pkg/frontend/querymiddleware/astmapper/sharding.go

```diff
@@ -542,12 +553,13 @@ func (summer *shardSummer) shardAndSquashBinOp(expr *parser.BinaryExpr) (parser.
 			return nil, err
 		}
 
-		children = append(children, &parser.BinaryExpr{
+		var binExpr parser.Expr = &parser.BinaryExpr{
```

Comment:
[nit] Same as above, can we use this simpler form?

```suggestion
		binExpr := &parser.BinaryExpr{
```